### PR TITLE
Move tests to its own package

### PR DESCRIPTION
**Pull request by nmuesch**
This PR aims to move the tests for this project into its own `test` sub folder/package. The hope is to make the structure of this repo a bit easier to parse through. 

While most of the PR is simply copying `*_test.go` files into a new `tests` folder, there are a couple of functional changes:
* Since the tests now reside in a different package than `datadog`, any code used from the main project needs to be imported
* Due to ^, this means there were a few helper methods we defined on resources that required being `Exported`. (These are helpers like: `translateClientError ` -> `TranslateClientError`, `GetUserAgent`, etc)
* The cassettes folder now lives under `tests` as well. 
* All filenames under `testFiles2EndpointTags` are now prefixed with `tests/`

### CODE REVIEW

**Review by pietdaniel — DISMISSED** [review]
approved for aaa code owner

**Review by DrkSephy — APPROVED** [review]
LGTM for Dashboards

**Review by zippolyte — DISMISSED** [review]

**Comment by nmuesch** [comment]
Apologies for everyone being notified on this one. The goal is to try and clean up the structure of the repo a bit since we're getting

**Review by zippolyte — DISMISSED** [review]

**Comment by nmuesch** [comment]
Few changes since the last commits/reviews (in addition to rebasing latest master)

Most notably: 
* Moved utils + validations to their own respective packages in `internal`
  * Moved the tests for these alongside their code in the `internal` package. The `make test` command still picks this up since its specifying the packages as `./...` wildcard
* Added a "Deprecated" comment to the utilities section of the `screenbaord` resource. It looks like that was previously used to place any and all utils.

File: datadog/resource_datadog_integration_aws.go [hunk 1]
```diff
@@ -6,13 +6,15 @@ import (
...
+	"github.com/terraform-providers/terraform-provider-datadog/datadog/internal/utils"
+
 	datadogV1 "github.com/DataDog/datadog-api-client-go/api/v1/datadog"
 	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
 )
 
 var integrationAwsMutex = sync.Mutex{}
 
-func accountAndRoleFromID(id string) (string, string, error) {
+func AccountAndRoleFromID(id string) (string, string, error) {
```
Comment (zippolyte) [inline datadog/resource_datadog_integration_aws.go:17]: shouldn't this be moved to utils to follow the pattern ?

File: datadog/resource_datadog_role.go [hunk 2]
```diff
@@ -90,32 +90,14 @@ func validatePermissionsUnrestricted(value interface{}, meta interface{}) error
 	return nil
 }
 
-func getRolePermissionSchema() *schema.Resource {
```
Comment (zippolyte) [inline datadog/resource_datadog_role.go:93]: This I'd keep here, it's for the schema it's interesting to have in the same file

File: datadog/tests/key_test.go [hunk 3]
```diff
@@ -1,8 +1,10 @@
-package datadog
```
Comment (zippolyte) [inline datadog/tests/key_test.go:1]: this and it's main file `key.go` should be moved to internal package

**Review by zippolyte — CHANGES_REQUESTED** [review]

File: datadog/resource_datadog_integration_aws.go (hunk 1 above)
Comment (nmuesch) [reply datadog/resource_datadog_integration_aws.go:17]: Good catch, I thought I moved this but I only moved the `AccountAndNamespaceFromID` function. 
Moving now

File: datadog/resource_datadog_role.go (hunk 2 above)
Comment (nmuesch) [reply datadog/resource_datadog_role.go:93]: Yeah thats fair. The schema is used in tests, so it'd have to be exported, but I agree it should live here.
Thinking more, the Schema is pretty much the external API of a provider, so 👍 to exporting.

**Review by zippolyte — APPROVED** [review]

**Pull request merged**